pmd-modelica/pom.xml

```diff
@@ -78,6 +78,27 @@
             <artifactId>pmd-core</artifactId>
         </dependency>
 
+        <dependency>
```

Comment:
TODO: Add comment, that the order matters.

Reply:
TODO: test adding these dependencies as plugin dependencies for surefire

Reply:
Ok, I figured out the problem now: If we have kotlintest-runner-junit5 as a dependency, this pulls in junit-platform 1.2.0, but for the junit-vintage-engine, we would need 1.5.x. Reversing the order, pulls in either junit-platform 1.2.x from kotlintest or junit-plaform 1.5.x from junit-vintage-engine. And surefire seems to need the newer version.

The correct way is, to define the two engines as plugin dependencies for surefire (see https://maven.apache.org/surefire/maven-surefire-plugin/examples/junit-platform.html), section "JUnit4 API in test dependencies": We have the Junit4 API (`@Test`, etc.) as a test-scope-dependency, but the engine is not available for tests - only for surefire, to run the tests.
We can add both kotlin and junit-vintage engines - but that only works with surefire-plugin 3+, so I've upgraded this plugin as well.

With that change, the order of the dependencies now doesn't matter anymore.

Removing kotlintest-runner-junit5 as a small issue: this not only provided the engine, but also part of the testing API (package `io.kotlintest.specs`), e.g. the base class `FunSpec`. This is now not longer available and I switched that to simply use `AbstractFunSpec` - which is available from io.kotlintest.core.
I tried to update io.kotlintest, but there are some incompatibilites (something to to with suspend/coroutines in the test framework), so I stayed at 3.1.8.
The io.kotlintest framework has been renamed with version 4.x to kotest: https://github.com/kotest/kotest - the upgrade will take a bit more effort. But in that version, `FunSpec` would be available in the core module, and the runner really only contains the engine...

@oowekyala : just FYI

Reply:
Ok thanks for figuring this out! I'll open a ticket to track the kotest upgrade